python/pylibcudf/pylibcudf/types.pyx

```diff
@@ -69,6 +70,48 @@ cdef class DataType:
         """Get the scale associated with this data type."""
         return self.c_obj.scale()
 
+    def _python_typecode(self) -> str:
```

Comment:
Could we make these `@property`'s so we can easily transition to `@cached_property` when available?

Reply:
Done https://github.com/rapidsai/cudf/pull/18941/commits/46a0755e5c9f81dad517975c55e53fde7fed2a0d